### PR TITLE
#416Enabled users to view public encounter metadata

### LINKED ISSUE #416: Public Encounter data is too restricted
**Expected behavior**
When viewing an encounter submitted by username "public" (`/encounters/encounter.jsp`), the following metadata is visible to any viewer, even when not signed in:
- Location description
- Location ID
- Encounter date
- Latitude/Longitude
- Photographer
- Submitter

**Current behavior**
Some metadata became more restricted after a security update. When viewing an encounter submitted by username "public" (`/encounters/encounter.jsp`), the following metadata is only visible to users who are signed into Wildbook. This can discourage citizen scientists from staying engaged if they can't see certain encounter details of public submissions.

**Context**
@goddesswarship discussed this with @holmbergius and determined that security updates made to Wildbooks in Feb 2024 introduced a bug that made public Encounter data more restricted. This is especially a blocker for Wildbooks with a large citizen scientist focus as these submitters are emailed when an individual they've encountered is seen again. Now they can't follow the link in their email to view the new public encounter details like they could before because it's blocked by Silo security.

**Community link**
https://community.wildme.org/t/privacy-settings-changed/2778

### PR DESCRIPTION
**Pull request by Veda-Gogineni**
PR fixes #416 

**Changes**
- "Public" user can now view the specified encounter metadata when not signed in.

### CODE REVIEW

**Comment by TanyaStere42** [comment]
@Veda-Gogineni Can you update your PR description? I need the description, reference number, and a brief overview of changes made (see template)

**Comment by Veda-Gogineni** [comment]
Sure, I will update PR description.

File: src/main/webapp/encounters/encounter.jsp [hunk 1]
```diff
@@ -674,8 +674,8 @@ var encounterNumber = '<%=num%>';
 
 				//let's see if this user has ownership and can make edits
       			boolean isOwner = ServletUtilities.isUserAuthorizedForEncounter(enc, request,myShepherd);
-            boolean isPublic = ServletUtilities.isEncounterOwnedByPublic(enc) && request.getUserPrincipal() != null; //should at least be logged in
```
Comment (naknomum) [inline src/main/webapp/encounters/encounter.jsp:677]: reading through this file, i think we actually need 2 value: one for something is _publicly displayable_ and one when something is _public and can be edited_. perhaps like this:
```
boolean isPublicViewable = ServletUtilities.isEncounterOwnedByPublic(enc);
boolean isPublicEditable = isPublic && (request.getUserPrincipal() != null);
```

then, below, you will use one or the other of these depending on whether the code is just for **showing values** (`isPublicViewable`) or for **allowing editing** of public data (`isPublicEditable`)

File: src/main/webapp/encounters/encounter.jsp [hunk 2]
```diff
@@ -674,8 +674,8 @@ var encounterNumber = '<%=num%>';
...
-      			pageContext.setAttribute("editable", (isOwner || isPublic) && CommonConfiguration.isCatalogEditable(context));
+            boolean isPublic = ServletUtilities.isEncounterOwnedByPublic(enc);
+            pageContext.setAttribute("editable", (isOwner) && CommonConfiguration.isCatalogEditable(context));
```
Comment (naknomum) [inline src/main/webapp/encounters/encounter.jsp:678]: for example, this `pageContext...` line should then use `(isOwner || isPublicEditable)`

File: src/main/webapp/encounters/encounter.jsp [hunk 3]
```diff
@@ -834,7 +834,7 @@ $(function() {
             <div class="col-xs-12 col-sm-6" style="vertical-align: top;padding-left: 10px;">
 
 <%-- START LOCATION --%>
-<% if ((isOwner || isPublic) && CommonConfiguration.isCatalogEditable(context)) { %>
+<% if ((isOwner && CommonConfiguration.isCatalogEditable(context)) || (isPublic && !CommonConfiguration.isCatalogEditable(context))) { %>
```
Comment (naknomum) [inline src/main/webapp/encounters/encounter.jsp:837]: this should be returned to original code, except `(isOwner || isPublicEditable)`

File: src/main/webapp/encounters/encounter.jsp [hunk 4]
```diff
@@ -879,7 +879,7 @@ else {
 
 
 <%
-if(isOwner && enc.getLocation()!=null){
+if((isOwner || isPublic) && enc.getLocation()!=null){
 %>
```
Comment (naknomum) [inline src/main/webapp/encounters/encounter.jsp:883]: this clause, since it is only _displaying_ a value, should be `(isOwner || isPublicViewable)`.

**Review by naknomum — CHANGES_REQUESTED** [review]
some change suggestions regarding viewing versus editing _public data_.
can follow up in discord as needed.

**Review by naknomum — APPROVED** [review]
looks good

**Pull request merged**